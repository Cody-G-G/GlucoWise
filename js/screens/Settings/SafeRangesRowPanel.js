'use strict';
import React, {Component} from 'react';
import {View, Text, TextInput} from 'react-native';
import SafeRangeButton from "./SafeRangeButton";
import styles from "./styles";
import isInputValid from '../../helpers/util/readingValidator';
import ReadingValueInput from "../../helpers/components/ReadingValueInput";

export default class SafeRangesRowPanel extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.safeRangesRowPanel}>
                <Text style={styles.safeRangeInputLabel}>{this.props.inputLabel}</Text>
                <ReadingValueInput style={styles.valueInput}
                                   inputValue={this.props.inputValue}
                                   onChangeText={this.props.updateSafeRange}/>
                <SafeRangeButton backgroundColor={'royalblue'}
                                 buttonText={'Set'}
                                 handlePress={() => {
                                     isInputValid(this.props.inputValue) ? this.props.saveSafeRange() : alert("Please input a valid blood glucose value");
                                 }}/>
                <SafeRangeButton backgroundColor={'darkorange'}
                                 buttonText={'Default'}
                                 handlePress={this.props.defaultSafeRange}/>
            </View>
        )
    }
}