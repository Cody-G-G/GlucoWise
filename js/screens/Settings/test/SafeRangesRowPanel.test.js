'use strict';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import SafeRangesRowPanel from "../SafeRangesRowPanel";

test('SafeRangesRowPanel - renders correctly', () => {
    const inputLabel = 'Min:';
    const inputValue = '80';
    const standard = 'mg/dL';

    const tree = renderer.create(
        <SafeRangesRowPanel inputLabel={inputLabel}
                            inputValue={inputValue}
                            standard={standard}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});