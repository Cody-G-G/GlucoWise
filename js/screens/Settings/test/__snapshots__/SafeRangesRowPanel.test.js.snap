// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeRangesRowPanel - renders correctly 1`] = `
<View
  style={
    Object {
      "borderWidth": 1,
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "flex": 1,
        "fontSize": 30,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Min:
  </Text>
  <TextInput
    keyboardType="numeric"
    maxLength={5}
    onChangeText={[Function]}
    style={
      Object {
        "color": "white",
        "flex": 1,
        "fontSize": 25,
      }
    }
    value="80"
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "royalblue",
        "borderWidth": 0.5,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
        }
      }
    >
      Set
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "darkorange",
        "borderWidth": 0.5,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
        }
      }
    >
      Default
    </Text>
  </View>
</View>
`;
